src/environment.js

```diff
@@ -0,0 +1,62 @@
+import {
+  isReactNative,
+  isWorker,
+  isWeb,
+  isNode,
+} from './services/environment';
+import {
+  FetchInterceptor,
+} from './fetchInterceptor';
+
+const interceptors = [];
+
+function init() {
+  if (isReactNative()) {
+    attach(global);
+  } else if (isWorker()) {
+    attach(self);
+  } else if (isWeb()) {
+    attach(window);
+  } else if (isNode()) {
+    attach(global);
+  } else {
+    throw new Error('Unsupported environment for fetch-token-intercept');
+  }
+}
+
+function attach(env) {
+  if (!env.fetch) {
+    throw Error('No fetch available. Unable to register fetch-token-intercept');
+  }
+
+  // for now add default interceptor
+  interceptors.push(new FetchInterceptor(env.fetch));
+
+  // monkey patch fetch
+  const fetchWrapper = fetch => (...args) => interceptors[0].intercept(...args);
```

Comment:
As we discussed, we're at a point that this could be implemented easily when needed, and we don't really need it. I believe it goes to 'nice to have' list when we settle on implementation.